receiver.go

```diff
@@ -316,6 +316,7 @@ func (r *Receiver) messageDisposition(ctx context.Context, msg *Message, state e
 	case err := <-wait:
 		// we've received confirmation of disposition
 		r.link.DeleteUnsettled(msg)
+		msg.settled = true
```

Comment:
Just being thorough or would I have seen possible errors because of this?

Reply:
We had a bug where if you tried to settle a message a second time it would hang.  I just realized I forgot to reference that bug, I added it to the comments.

Reply:
Cool, makes sense.